### DISCUSSION TITLE
How to output equivalent_plastic_strain and elastic strain energy?

### DISCUSSION THREAD

**Discussion by RWTHLHK** [idaholab/moose#26212] [Q&A Modules: Solid mechanics]
I need to output equivalent_plastic_strain and elastic strain energy in my app. Is it possible to output them directly or I have to compute them by myself?

**Comment by ngrilli** [comment]
@RWTHLHK  you can output and visualize in paraview any material property declared in the material object.

For instance I use this material:

https://github.com/idaholab/moose/blob/next/modules/tensor_mechanics/src/materials/ComputePFFractureStressBase.C

I add to the input file an auxvariable:

  [./elastic_energy]
    order = CONSTANT
    family = MONOMIAL
  [../]

then an auxkernel:

  [./elastic_energy]
    type = MaterialRealAux
    variable = elastic_energy
    property = elastic_energy
  [../]

then you will be able to visualize it in paraview. Same for other material properties.

**Reply by RWTHLHK** [reply]
Thanks a lot! I also solve the problem by setting declared property to output properties in a material project then adding a Postprocessor SideAverageMaterialProperty or NodeAverageMaterialProperty, etc. for postprocessing.
